### DISCUSSION TITLE
[Feature] <ignore thumbnails in existing library>

### DISCUSSION THREAD

**Discussion by CharkLu** [immich-app/immich#3064] [Feature Request]
### The feature

Some of my existing photo library contains thumbnails generated by my NAS. While importing an existing library through CLI tool these thumbnails was also imported, thus resulting multiple duplicate image in photo library. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by etnoy** [comment]
This is something I'm planning to work on myself in the future. Can you tell me more about your directory structure?

**Reply by CharkLu** [reply]
Here are my directory structure:

<pre>

└── Photos/            
    ├── 2021/
    │   └── 2021-01-01.jpg
    ├── 2022/
    │   └── 2022-01-01.jpg     
    ├── 2023-01-01.heic
    ├── 2023-01-01.mov
    ├── 2023-01-02.jpg     
    ├── @Recycle (recycle folder created and managed by QNAP)
    ├── .@__thumb (thumbnails folder created and managed by QNAP)
    └── .DS_Store</pre>

**Reply by etnoy** [reply]
Hey, if you feel like getting your hands dirty a little, the new CLI I've written support ignoring file paths: https://github.com/immich-app/immich/pull/3066

**Reply by CharkLu** [reply]
wow, this feature is already implemented. Great!

**Reply by indam** [reply]
@etnoy Sorry for disturbing, but I also encountered similar confusion. I manage the photos with FileRun, it generates thumbnails folder under each directory, like /.filerun.thumbnails/. How can I ignore them, when importing photos with immich cli? Or where to put a ignore file to skip them? Please help. Thank you in advance.

**Reply by etnoy** [reply]
Try the new cli under the cli folder in the main repo, it is not fully finished but has an ignore flag

**Reply by indam** [reply]
Thank you for your reply, I just noticed that I was using the `immich` command pre-installed inside the immich-server container, which are totally different.

**Reply by haldi4803** [reply]
Side note.
Synology NAS uses the `@eaDir` folder for all thumbnails.

**Comment by mmomjian** [comment]
We now support exclusion paths for external libs.
